src/table.ts

```diff
@@ -16,38 +16,34 @@
 
 'use strict';
 
-import {promisifyAll} from '@google-cloud/promisify';
+import { promisifyAll } from '@google-cloud/promisify';
 import * as is from 'is';
-import {ServiceError} from 'grpc';
+import { ServiceError } from 'grpc';
 import * as through from 'through2';
+import { Operation as GaxOperation } from 'google-gax/build/src/longrunning';
 
-import {Json} from './codec';
-import {Database} from './database';
-import {DatabaseAdminClient as d, SpannerClient as s} from './v1';
-import {PartialResultStream, Row} from './partial-result-stream';
-import {Snapshot, ReadRequest, Transaction, TimestampBounds} from './transaction';
+import { Json } from './codec';
+import { Database } from './database';
+import { PartialResultStream, Row } from './partial-result-stream';
+import { ReadRequest, TimestampBounds, Transaction, Snapshot } from './transaction';
+import { Schema, LongrunningIOperation, RequestCallback, ICommitResponse } from './common';
 
-export type Key = string|string[];
+export type Key = string | string[];
+export type CreateTableResponse = [Table, GaxOperation, LongrunningIOperation];
+export type CommitCallback = RequestCallback<ICommitResponse>;
+export type CommitResponse = [ICommitResponse];
+export type DeleteResponse = [GaxOperation, LongrunningIOperation];
 
-type Schema = string|string[]|{statements: string[], operationId?: string};
-
-type CommitPromise = Promise<[s.CommitResponse]>;
-type CreateTablePromise = Promise<[Table, d.Operation, d.GrpcOperation]>;
-type DropTablePromise = Promise<[d.Operation, d.GrpcOperation]>;
-type ReadPromise = Promise<[Array<Row|Json>]>;
-
-interface CreateTableCallback {
+export interface CreateTableCallback {
   (err: ServiceError, table?: null, operation?: null, apiResponse?: null): void;
-  (err: null, table: Table, operation: d.Operation,
```

Comment:
This isn't anything you did, but I recently learned that interfaces with overloads (like these) don't infer types properly so I think we actually want to avoid them. Might be worthwhile to create a generic for callbacks in `common.ts`.

Reply:
Great to know, will do!